### PR TITLE
Added support for typescript.

### LINKED ISSUE #177: Set up Typescript Support
# Background

We want to transition to typescript, so we need to add support for it. 

# Task

* Add typescript support

# Done When

* New files can be written in Typescript

### PR DESCRIPTION
**Pull request by lullofthesea**
Closes #177 

Added packages for typescript / types for express and node.  Added tsconfig.json

Tested by temporarily changing app.js and www.js to .ts files and seeing if type errors were thrown (they were).

### CODE REVIEW

**Review by aidenszeto — APPROVED** [review]

**Review by nishantbalaji — APPROVED** [review]

**Comment by aidenszeto** [comment]
lets wait until all PRs from last sprint are merged before merging this (just pending Auth refactor PRs)

**Comment by aidenszeto** [comment]
release done, this is good for a merge whenever you're ready

**Pull request merged**
